client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClient.java

```diff
@@ -47,6 +47,15 @@
      */
     abstract boolean compareAndSetAttribute(SegmentAttribute attribute, long expectedValue, long newValue);
     
+    /**
+     * Deletes all data before offset of the provided segment.
+     * This data will no longer be readable. Existing offsets are not affected by this operations. 
+     * The new startingOffset will be reflected in {@link SegmentMetadataClient#getSegmentInfo().startingOffset}.
+     * @param segment The segment to truncate.
+     * @param offset The offset the segment should be truncated at.
+     */
+    abstract void truncateSegment(Segment segment, long offset);
```

Comment:
This isn't really just a metadata operations as it is deleting actual data of the segment. I suppose that you're proposing we do it here for lack of a better place to do it?

I'm also wondering whether this conflicts with any of the truncation that the controller is doing. The implementation of the class is internal and only used with the revisioned stream client, so it is not really truncating segments of streams visible to applications. Could you comment on whether my assessment makes sense and whether you see any potential issue?

Reply:
Per our discussion, delaying this until: https://github.com/pravega/pravega/issues/2197